RT/RTmaterials.c

```diff
@@ -154,43 +154,82 @@ static int RT_LoadMaterialTexturesFromPaths(uint16_t bm_index, RT_Material *mate
 	{
 		if (load_mask & RT_BITN(i))
 		{
+			
+			// first try loading the compressed dds version of the file if it exists
+			bool ddsLoaded = false;
 			RT_ArenaMemoryScope(&g_thread_arena)
 			{
-				char *file = RT_ArenaPrintF(&g_thread_arena, "assets/textures/%s", paths->textures[i]);
+				char* ddsFile = RT_ArenaPrintF(&g_thread_arena, "assets/textures/%s.dds", paths->textures[i]);
+				
+				uint8_t* ddsData = NULL;  // this is pointer to whole file including header
+				const struct DDS_HEADER* header = NULL;	// pointer to the header (in same buffer as ddsData)
+				const uint8_t* bitData = NULL;		// pointer to the compressed pixel data (in same buffer as ddsData)
+				size_t bitSize = 0;					// size of the compressed pixel data
+				
+				bool fileLoaded = RT_LoadDDSImageFromDisk(&g_thread_arena, ddsFile, &ddsData, &header, &bitData, &bitSize);
+
+				if (fileLoaded && ddsData)
+				{
+					ddsLoaded = true;
+
+					material->textures[i] = RT_UploadTextureDDS(&(RT_UploadTextureParamsDDS) {
+						.header = header,
+						.ddsData = ddsData,
+						.bitData = bitData,
+						.bitSize = bitSize,
+						.sRGB = i == 0 || i == 4,	// set sRGB to true if basecolor or emissive
+						.name = RT_ArenaPrintF(&g_thread_arena, "Game Texture %hu:%s (source: %s)",
+								bm_index, g_rt_texture_slot_names[i], ddsFile)
+					});
 
-				RT_TextureFormat format = g_rt_material_texture_slot_formats[i];
+					textures_reloaded += 1;
 
-				int bpp = g_rt_texture_format_bpp[format];
+					free (ddsData);		// free buffer holding dds data.  Don't need to free the other pointers as they simply point to other adddresses in this memory block.
```

Comment:
if the arena change on malloc is implemented the free is not needed anymore as RT_ArenaMemoryScope will free it automatically.